### PR TITLE
Add full integration test DAG

### PR DESCRIPTION
**Pull request by dimberman**
To ensure full feature parity between data systems, we should develop integration DAGs that test all known scenarios, this is beyond just example DAGs as the DAGs will make it easy to switch databases by changing the input table params.

### CODE REVIEW

File: src/astro/sql/__init__.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def append(
         columns=columns,
         casted_columns=casted_columns,
         **kwargs,
-    )
+    ).output
```
Comment (tatiana) [inline src/astro/sql/__init__.py:92]: @dimberman why we didn't have `output` before? Did it work?

File: tests/integration_test_dag.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    elif isinstance(hook, PostgresHook):
+        return TempTable(conn_id=hook.postgres_conn_id, database=hook.schema)
+    elif isinstance(hook, SqliteHook):
+        return TempTable(conn_id=hook.sqlite_conn_id, database="sqlite")
+    # elif isinstance(hook, BigQueryHook):
+    #     return TempTable(conn_id=hook.gcp_conn_id, database=)
+
+
+@pytest.fixture
+def merge_keys(sql_server):
```
Comment (tatiana) [inline tests/integration_test_dag.py:79]: We might be overusing Pytest fixtures here - would it be possible to use a normal function?

File: tests/integration_test_dag.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    sql_name, _ = sql_server
+
+    if sql_name == "snowflake":
+        return {"list": "list", "sell": "sell"}
+    else:
+        return ["list", "sell"]
+
+
+@aql.transform
+def do_a_thing(input_table: Table):
```
Comment (tatiana) [inline tests/integration_test_dag.py:95]: We could give more meaningful names to the functions, e.g.:
`apply_transform`

File: tests/integration_test_dag.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
+import pathlib
```
Comment (tatiana) [inline tests/integration_test_dag.py:1]: Did you consider also adding `truncate`. `render` and `stats_check` the `integration_test_dag.py`?

**Review by tatiana — APPROVED** [review]
This is brilliant, @dimberman. :clap:  :clap: :clap: 

It will give us much more confidence in keeping things working in Astro.

I made some questions, but they are not a blocker for approving this ticket. We can have to follow up on dedicated issues to address them.

File: tests/integration_test_dag.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        return TempTable(
+            conn_id=hook.snowflake_conn_id,
+            database=hook.database,
+            warehouse=hook.warehouse,
+        )
+    elif isinstance(hook, PostgresHook):
+        return TempTable(conn_id=hook.postgres_conn_id, database=hook.schema)
+    elif isinstance(hook, SqliteHook):
+        return TempTable(conn_id=hook.sqlite_conn_id, database="sqlite")
+    # elif isinstance(hook, BigQueryHook):
```
Comment (tatiana) [inline tests/integration_test_dag.py:74]: Shall we have a ticket to re-enable this?

File: src/astro/sql/__init__.py (hunk 1 above)
Comment (dimberman) [reply src/astro/sql/__init__.py:92]: @tatiana without that `.output` there is no way to pass the result of the append to a validation function. Before the class was simply returning nothing. I'll make a ticket for this feature add, but this won't break any existing workflows.

File: tests/integration_test_dag.py (hunk 4 above)
Comment (dimberman) [reply tests/integration_test_dag.py:1]: @tatiana I want to add those in future PRs. I want to start with a working integration test DAG and then as we fix the test cases we will add all of those features in.

File: tests/integration_test_dag.py (hunk 2 above)
Comment (dimberman) [reply tests/integration_test_dag.py:78]: Fair enough, converted to function :)

File: tests/integration_test_dag.py (hunk 3 above)
Comment (dimberman) [reply tests/integration_test_dag.py]: Done

File: tests/integration_test_dag.py (hunk 5 above)
Comment (dimberman) [reply tests/integration_test_dag.py]: Bigquery now works with recent merges! We'll just need to keep checking as we add new features to this DAG.

**Pull request merged**